### PR TITLE
Create button to calculate stooip PEX-20

### PR DESCRIPTION
**Pull request by FreddyEcu-Ch**
Create button to calculate stooip PEX-20

### CODE REVIEW

File: interfaz_controller/interfaz_controller.py [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-
-@xw.func
-def hello(name):
-    return f"Hello {name}!"
+    area = float(sheet['B4'].value)
+    h = float(sheet['B5'].value)
+    poro = float(sheet['B6'].value)
+    sw = float(sheet['B7'].value)
+    boi = float(sheet['B8'].value)
+    sheet['B10'].value = 7758 * area * h * poro * (1 - sw) / boi
```
Comment (zeratul-3011) [inline interfaz_controller/interfaz_controller.py]: I don't see how can I calculate this within the excel file. You need to add a button for it

File: interfaz_controller/interfaz_controller.py [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
-
-
-@xw.func
-def hello(name):
-    return f"Hello {name}!"
+    area = float(sheet['B4'].value)
+    h = float(sheet['B5'].value)
+    poro = float(sheet['B6'].value)
+    sw = float(sheet['B7'].value)
+    boi = float(sheet['B8'].value)
```
Comment (zeratul-3011) [inline interfaz_controller/interfaz_controller.py]: It is better to load all this data at once using a numpy array option or dataframe converter. If you do this, excel calls xlwings for each cell which is inefficient.

**Review by zeratul-3011 — CHANGES_REQUESTED** [review]
Please review the comments and make appropriate changes to merge the PR

File: interfaz_controller/interfaz_controller.py (hunk 1 above)
Comment (FreddyEcu-Ch) [reply interfaz_controller/interfaz_controller.py]: image

It was added a button

Comment (FreddyEcu-Ch) [reply interfaz_controller/interfaz_controller.py]: Ifyou hit it, it calculates the Stoiip

Comment (zeratul-3011) [reply interfaz_controller/interfaz_controller.py]: I don't see a button in the excel file. Did you push it? @FreddyEcu-Ch
image

File: interfaz_controller/interfaz_controller.py (hunk 2 above)
Comment (FreddyEcu-Ch) [reply interfaz_controller/interfaz_controller.py]: Done

File: interfaz_controller/interfaz_controller.py [hunk 3]
```diff
@@ -1,21 +1,17 @@
...
-    if sheet["A1"].value == "Hello xlwings!":
-        sheet["A1"].value = "Bye xlwings!"
-    else:
-        sheet["A1"].value = "Hello xlwings!"
 
-
-@xw.func
-def hello(name):
-    return f"Hello {name}!"
+    params = sheet['B4:B8'].options(np.array, transpose=True).value
```
Comment (zeratul-3011) [inline interfaz_controller/interfaz_controller.py]: Awesome! Now, what happens if the user of the excel sheet wants to add a row at the top of the file? Will this code work? Probably not, that's why I suggest using named ranges within excel, and then call the name of that range within xlwings. In that way you will make sure that range maintains no matter what the user does.

**Review by zeratul-3011 — CHANGES_REQUESTED** [review]

File: interfaz_controller/interfaz_controller.py (hunk 3 above)
Comment (FreddyEcu-Ch) [reply interfaz_controller/interfaz_controller.py]: Done

Comment (zeratul-3011) [reply interfaz_controller/interfaz_controller.py]: Sorry, it is not ok. When I add a row in excel the calculation throws an error which means that the range is not persisted. Try using the Name Manager inside excel to assign a name to the range. Don't do it within the code.

File: interfaz_controller/interfaz_controller.py [hunk 4]
```diff
@@ -6,7 +6,8 @@ def main(workbook: xw.Book = None):
     wb = xw.Book.caller()
     sheet = wb.sheets[0]
 
-    params = sheet['B4:B8'].options(np.array, transpose=True).value
+    sheet.range('B4:B8').name = 'values'
```
Comment (zeratul-3011) [inline interfaz_controller/interfaz_controller.py]: Not doing what is intented. When adding a row in excel the calculation throws an error.

**Review by zeratul-3011 — CHANGES_REQUESTED** [review]

File: interfaz_controller/interfaz_controller.py (hunk 3 above)
Comment (FreddyEcu-Ch) [reply interfaz_controller/interfaz_controller.py]: Yes, you are right, now it was corrected

File: interfaz_controller/interfaz_controller.py [hunk 5]
```diff
@@ -1,21 +1,18 @@
...
-    else:
-        sheet["A1"].value = "Hello xlwings!"
 
-
-@xw.func
-def hello(name):
-    return f"Hello {name}!"
+    # Cells B4 to B8 are called with the name Ranges, defined on the name manager from Ms Excel
+    params = sheet['Ranges'].options(np.array, transpose=True).value
+    sheet['B10'].value = 7758 * params[0] * params[1] * params[2] * (1 - params[3]) / params[4]
```
Comment (zeratul-3011) [inline interfaz_controller/interfaz_controller.py]: The cell that displays the value of STOIIP should also have a named range. When I add a row in excel, the result now calculates correctly but in another cell. Correct this please

**Review by zeratul-3011 — CHANGES_REQUESTED** [review]

File: interfaz_controller/interfaz_controller.py (hunk 5 above)
Comment (FreddyEcu-Ch) [reply interfaz_controller/interfaz_controller.py]: Done

**Review by zeratul-3011 — APPROVED** [review]
Good!

**Pull request merged**